src/Repository/RepositoryStack.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Repository;
+
+/**
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+class RepositoryStack
```

Comment:
We don't really need a "stack", do we? We will only every be dealing with *one* repository at a time. What about `RepositoryContext`, with a `setCurrentRepository` and `getCurrentRepository()`? If there is no current repository and someone calls `getCurrentRepository`, we should throw an exception (as that's probably a developer error).